### PR TITLE
Merge main into feat/sanchonet

### PR DESCRIPTION
**Pull request by przemyslaw-wlodek**
# Checklist

- [x] JIRA - [LW-9215](https://input-output.atlassian.net/browse/LW-9215)

---

## Testing
Please re-test the transaction confirmation process, mainly Dapp transactions, including new Certificates and Voting Procedures.

[LW-9215]: https://input-output.atlassian.net/browse/LW-9215?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/ui/components/ActivityDetail/TransactionFee.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const { t } = useTranslate();
+
+  return (
+    <div className={styles.details}>
+      <div className={styles.txFeeContainer}>
+        <div className={styles.txfee} data-testid="tx-fee-title">
+          {t('package.core.activityDetails.transactionFee')}
+        </div>
+        <Tooltip title={t('package.core.activityDetails.transactionFeeInfo')}>
+          {Info ? (
```
Comment (vetalcore) [inline packages/core/src/ui/components/ActivityDetail/TransactionFee.tsx:24]: would it ever be false?

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx [hunk 2]
```diff
@@ -1,91 +1,84 @@
...
+  fiatCurrencyPrice,
+  coinSymbol
+}: DappTransactionProps): React.ReactElement => {
+  const { t } = useTranslate();
+  return (
+    <div>
+      <DappInfo {...dappInfo} className={styles.dappInfo} />
+      {errorMessage && <ErrorPane error={errorMessage} className={styles.error} />}
+      <div data-testid="dapp-transaction-container" className={styles.details}>
+        {type === 'Mint' && mintedAssets?.length > 0 && (
```
Comment (vetalcore) [inline packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:39]: could we do something like:
```suggestion
        {type === Wallet.Cip30TxType.Mint && mintedAssets?.length > 0 && (
```

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx [hunk 3]
```diff
@@ -1,91 +1,84 @@
...
+            <DappTxHeader
+              title={t('package.core.dappTransaction.transaction')}
+              subtitle={t('package.core.dappTransaction.mint')}
+            />
+            {mintedAssets.map((asset) => (
+              <DappTxAsset key={asset.name} {...asset} />
+            ))}
+          </>
+        )}
+        {type === 'Burn' && burnedAssets?.length > 0 && (
```
Comment (vetalcore) [inline packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:50]: ditto

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx [hunk 4]
```diff
@@ -1,91 +1,84 @@
...
+            <DappTxHeader
+              title={mintedAssets?.length > 0 ? undefined : t('package.core.dappTransaction.transaction')}
+              subtitle={t('package.core.dappTransaction.burn')}
+            />
+            {burnedAssets.map((asset) => (
+              <DappTxAsset key={asset.name} {...asset} />
+            ))}
+          </>
+        )}
+        {type === 'Send' && (
```
Comment (vetalcore) [inline packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:61]: ditto

File: packages/staking/src/features/activity/EpochsSwitch.module.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.buttonsBackground {
+  background-color: var(--light-mode-light-grey, var(--dark-mode-dark-grey, #2f2f2f));
+  border-radius: 1rem;
```
Comment (vetalcore) [inline packages/staking/src/features/activity/EpochsSwitch.module.scss:3]: are we sure it should be dependant on the font size?

File: packages/staking/src/features/activity/EpochsSwitch.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { ControlButton, Flex, Text } from '@lace/ui';
+import { useTranslation } from 'react-i18next';
+import styles from './EpochsSwitch.module.scss';
+
+// eslint-disable-next-line no-magic-numbers
+const EPOCHS_OPTIONS = [5, 15];
```
Comment (vetalcore) [inline packages/staking/src/features/activity/EpochsSwitch.tsx:6]: are those available somewhere in the sdk/typings?

File: packages/staking/src/features/activity/PastEpochsRewards/PastEpochsRewards.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { Flex, Text } from '@lace/ui';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { EpochsSwitch } from '../EpochsSwitch';
+import { useRewardsByEpoch } from './hooks/useRewardsByEpoch';
+import { RewardsChart } from './RewardsChart';
+
+const DEFAULT_LAST_EPOCHS = 5;
```
Comment (vetalcore) [inline packages/staking/src/features/activity/PastEpochsRewards/PastEpochsRewards.tsx:8]: ditto

**Review by vetalcore — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/components/MainMenu/DropdownMenuOverlay/components/UserInfo.tsx [hunk 8]
```diff
@@ -44,29 +47,51 @@ export const UserInfo = ({ avatarVisible = true }: UserInfoProps): React.ReactEl
...
-                <p className={styles.walletName} data-testid="header-menu-wallet-name">
-                  {walletName}
-                </p>
-                <p className={styles.walletAddress} data-testid="header-menu-wallet-address">
-                  {handleName || shortenedWalletAddress}
-                </p>
+          {process.env.USE_MULTI_WALLET === 'true' ? (
+            <ProfileDropdown.WalletOption
+              title={walletInfo.name}
+              subtitle="Account #0"
```
[the remark this answers is not in the document]
Comment (przemyslaw-wlodek) [reply apps/browser-extension-wallet/src/components/MainMenu/DropdownMenuOverlay/components/UserInfo.tsx:60]: Nice catch! Asked @renanvalentin to take a look into it ([PR](https://github.com/input-output-hk/lace/pull/742/files#diff-01d7a2b53a4ac7faaa670de72313daa0e7ec3699717d36708f55db1d22a052faR53))

File: apps/browser-extension-wallet/src/features/account/components/EditAccount/EditAccountDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+export const EditAccountDrawer = ({ name, index, visible, onSave, hide }: Props): React.ReactElement => {
+  const { t } = useTranslation();
+  const [currentName, setCurrentName] = useState(name);
+
+  return (
+    <Drawer
+      zIndex={999}
+      open={visible}
+      navigation={<DrawerNavigation title={name || `Account #${index}`} onCloseIconClick={hide} />}
```
[the remark this answers is not in the document]
Comment (przemyslaw-wlodek) [reply apps/browser-extension-wallet/src/features/account/components/EditAccount/EditAccountDrawer.tsx:22]: Again nice catch 😄 Asked @lucas-barros to fix it ([PR](https://github.com/input-output-hk/lace/pull/746/files#diff-863904267e5a691f4cb041c87b983ce15b325df8679f98ed5d540211110acebcR22))

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx (hunk 3 above)
Comment (przemyslaw-wlodek) [reply packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:50]: I agree but this was already approved. We're going to refactor `DappTransaction` very soon, so I'd advise to keep it for now as it is. 🙏🏻

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx (hunk 4 above)
Comment (przemyslaw-wlodek) [reply packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:61]: Same [as above](https://github.com/input-output-hk/lace/pull/763#discussion_r1411936759)

File: packages/core/src/ui/components/DappTransaction/DappTransaction.tsx (hunk 2 above)
Comment (przemyslaw-wlodek) [reply packages/core/src/ui/components/DappTransaction/DappTransaction.tsx:39]: Good catch. I overlooked it during migration. We have so much duplication of types across extension and core/cardano packages. Let's improve it.

**Review by vetalcore — APPROVED** [review]

**Pull request merged**